codalab/model/bundle_model.py

```diff
@@ -861,17 +861,19 @@ def transition_bundle_staged(self, bundle):
             ).fetchone()
             if not row:
                 raise IntegrityError('Missing bundle with UUID %s' % bundle.uuid)
-            if row.state != State.STARTING:
-                # It is possible that this method is called on a bundle
-                # that has started running.
-                return False
 
-            update_message = {'state': State.STAGED, 'metadata': {'job_handle': None}}
-            self.update_bundle(bundle, update_message, connection)
+            # Reset all metadata fields that aren't input by user from RunBundle class to be None.
+            # Excluding all the fields that can be set by users, e.g. the "action" field.
```

Comment:
"set by users, which for now is just the `action` field"

Reply:
Updated.